diff --git a/org.kde.kasts.desktop b/org.kde.kasts.desktop
index 590291d..ee78b6d 100644
--- a/org.kde.kasts.desktop
+++ b/org.kde.kasts.desktop
@@ -99,10 +99,16 @@ Keywords[uk]=трансляція
 Keywords[x-test]=xxpodcastxx
 Keywords[zh_CN]=播客
 Icon=kasts
-Exec=kasts
+Exec=qt-runner /usr/bin/kasts
 Type=Application
 Categories=Network;Qt;KDE;Feed;
 Terminal=false
 InitialPreference=10
 X-KDE-FormFactor=desktop;tablet;handset;
 StartupNotify=true
+
+-Nemo-Single-Instance=no
+X-Nemo-Application-Type=no-invoker
+
+[X-Sailjail]
+Sandboxing=Disabled
